app/gui/src/project-view/components/ComponentBrowser/input.ts

```diff
@@ -271,7 +282,13 @@ export function useComponentBrowserInput(
   }
 
   function applySourceNode(text: string) {
-    return sourceNodeIdentifier.value ? `${sourceNodeIdentifier.value}.${text}` : text
+    return (
+      sourceNodeIdentifier.value ?
+        /^[a-zA-Z]/.test(text) ?
```

Comment:
Technically an identifier can start with a character other than ASCII letters; we could handle that by exposing to WASM an entry point to the lexer so that we can see if the first token is a (lexical) identifier.

I suppose this is ok for an initial implementation but let's add a TODO.

Reply:
Well, implementation in rust should be quite easy; I just assumed this is too simple case for it.

But here you pointed me that it's not _that_ simple case.